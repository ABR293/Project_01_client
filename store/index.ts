import {createWrapper, Context, MakeStore, HYDRATE} from 'next-redux-wrapper';
import { Store } from 'redux'
import {configureStore} from '@reduxjs/toolkit'
import createSagaMiddleware, {Task} from 'redux-saga'
import { combineReducers } from '@reduxjs/toolkit'
import playerReducer  from './playerSlicer';
import tracksReducer  from './tracksSlicer';
import rootSaga from './sagas';

 const rootReducer = combineReducers({
    player: playerReducer,
    tracks: tracksReducer
})

export type RootState = ReturnType<typeof rootReducer>


export interface SagaStore extends Store {
  sagaTask?: Task;
}

export const reducer = (state:any, action: any) => {
  if (action.type === HYDRATE) {
    const nextState = {
      ...state, // use previous state
      ...action.payload, // apply delta from hydration
    };
    if (state.count) nextState.count = state.count; // preserve count value on client side navigation
    return nextState;
  } else {
    return rootReducer(state, action);
  }
};

let sagaMiddleware = createSagaMiddleware()

export const store = configureStore({
    reducer,
    middleware: [sagaMiddleware],
})

const makeStore = (context: Context) => {
  (store as SagaStore).sagaTask = sagaMiddleware.run(rootSaga);
  return store
}

export const wrapper = createWrapper<Store<RootState>>(makeStore, {debug: true}); 

// sagaMiddleware.run(rootSaga)


//  middleware: (getDefaultMiddleware) =>
//     getDefaultMiddleware().concat(middleware),

