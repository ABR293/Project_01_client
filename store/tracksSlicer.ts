import { ITrack } from "../types/tracks"

import { createSlice } from '@reduxjs/toolkit'
import type { PayloadAction } from '@reduxjs/toolkit'

const initialState = {
    query: '' as string | undefined,
    error: '',
    tracks: [] as ITrack[],
}
export type tracksState = typeof initialState

const trackSlice = createSlice({
    name: 'tracks',
    initialState,
    reducers: {
        // (state, action: PayloadAction<string> ){
        //   if(action.payload){
        //     state.query = action.payload
        //   }
        // },
        // setQuery(state, action: PayloadAction<string> ){
        //   state.query = action.payload
        // },
        fetchTracksSuccess(state, action: PayloadAction<ITrack[]> ){
          state.tracks = action.payload 
        },
        fetchTracksError(state, action: PayloadAction<string> ){
          state.error = action.payload 
        },
    }
})

export const trackActions = trackSlice.actions
export default trackSlice.reducer