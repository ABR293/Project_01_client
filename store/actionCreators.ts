

import {playerActions} from './playerSlicer'
import {trackActions} from './tracksSlicer'

const actionCreators = {
    ...playerActions,
    ...trackActions,
}

export default actionCreators