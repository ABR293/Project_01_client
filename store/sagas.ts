import { call, put, takeEvery, takeLatest } from 'redux-saga/effects'
import actionCreators from './actionCreators'
import axios, { Axios, AxiosResponse } from 'axios'
import { ITrack } from '../types/tracks';
import { ServerResponse } from 'http';
import Api from '../api';

function* workerFetchTracs({payload = ''}) {
   try {
      let data:ITrack[] = []
      if (payload !== '') {
         data = yield call(() => Api.searchTracks(payload));
      } else {
         data = yield call(() => Api.getAllTracks() );
      }
      yield put(actionCreators.fetchTracksSuccess(data));
   } catch (err) {
      yield put(actionCreators.fetchTracksError('Произошла ошибка при загрузке треков')); 
   }
}

function* tracksSaga() {
   // yield takeEvery('tracks/fetchTracks', workerFetchTracs)
   yield takeLatest('getTracks', workerFetchTracs)
}

export default  function* rootSaga() {
   yield tracksSaga()
}

