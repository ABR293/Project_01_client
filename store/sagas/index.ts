import { all } from "redux-saga/effects";
import { tracksSaga } from "./tracksSaga";
import authSaga from "./authSaga";
import { TypeOfSagaActyonType } from "../actionCreators";

export type SagaAcceptor = (A: TypeOfSagaActyonType, G: Function) => void;

export default function* rootSaga() {
  yield all([tracksSaga(), authSaga()]);
}
