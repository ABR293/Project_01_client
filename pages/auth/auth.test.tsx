import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import SignIn from "../../components/signIn";
import SignUp from "../../components/signUp";
import PasswordReset from "./passwordReset";

import { store } from "../../store/index";

jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: "/",
      pathname: "",
      query: "",
      asPath: "",
      push: jest.fn(),
    };
  },
}));

global.Audio = jest.fn().mockImplementation(() => ({
  pause: jest.fn(),
  play: jest.fn(),
}));

describe("Render", () => {
  it("should render SignIn screen", () => {
    render(
      <Provider store={store}>
        <SignIn onChangeReg={() => {}} />
      </Provider>
    );
    expect(screen.getByTestId("signIn")).toMatchSnapshot();
    //   screen.debug;
  });

  it("should render SignUp screen", () => {
    render(
      <Provider store={store}>
        <SignUp onChangeReg={() => {}} />
      </Provider>
    );
    expect(screen.getByTestId("signUp")).toMatchSnapshot();
    //   screen.debug;
  });

  it("should render PasswordReset screen", () => {
    render(
      <Provider store={store}>
        <PasswordReset />
      </Provider>
    );
    expect(screen.getByTestId("password-reset")).toMatchSnapshot();
    //   screen.debug;
  });
});
