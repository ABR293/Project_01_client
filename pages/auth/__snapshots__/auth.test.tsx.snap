// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render PasswordReset screen 1`] = `
<main
  class="MuiContainer-root MuiContainer-maxWidthXs css-nzinn5-MuiContainer-root"
  data-testid="password-reset"
>
  <div
    class="MuiBox-root css-binzgt"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1999axt-MuiAvatar-root"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="LockOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
        />
      </svg>
    </div>
    <h1
      class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
    >
      Reset Your Password
    </h1>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAccordion-root Mui-expanded css-1y725y-MuiPaper-root-MuiAccordion-root"
    >
      <div
        hidden=""
      >
        <div
          aria-controls="panel1d-content"
          aria-expanded="true"
          class="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded css-1pvvkxv-MuiButtonBase-root-MuiAccordionSummary-root"
          id="panel1d-header"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content Mui-expanded css-1betqn-MuiAccordionSummary-content"
          />
        </div>
      </div>
      <div
        class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
          >
            <div
              class="MuiAccordion-region"
              role="region"
            >
              <div
                class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
              >
                <form
                  class="MuiBox-root css-164r41r"
                  novalidate=""
                >
                  <div
                    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="false"
                      for="login"
                      id="login-label"
                    >
                      Email Address
                      <span
                        aria-hidden="true"
                        class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                      >
                         
                        *
                      </span>
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-invalid="false"
                        autocomplete="email"
                        class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                        id="login"
                        name="login"
                        required=""
                        type="text"
                        value=""
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-yjsfm1"
                        >
                          <span>
                            Email Address
                             
                            *
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-26yjs-MuiTypography-root"
                  >
                     
                  </p>
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1vhaqj4-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="submit"
                  >
                    <div
                      class="MuiBox-root css-1e16d4q"
                    >
                      Send e-mail With secret code
                    </div>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAccordion-root css-1y725y-MuiPaper-root-MuiAccordion-root"
    >
      <div
        hidden=""
      >
        <div
          aria-controls="panel2d-content"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root css-1pvvkxv-MuiButtonBase-root-MuiAccordionSummary-root"
          id="panel2d-header"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
          />
        </div>
      </div>
      <div
        class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
          >
            <div
              class="MuiAccordion-region"
              role="region"
            >
              <div
                class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
              >
                <form
                  class="MuiBox-root css-164r41r"
                  novalidate=""
                >
                  <h1 />
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    Sdcret Code was sended On your e-mail
                  </p>
                  <div
                    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="false"
                      for="password"
                      id="password-label"
                    >
                      Password
                      <span
                        aria-hidden="true"
                        class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                      >
                         
                        *
                      </span>
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-invalid="false"
                        autocomplete="current-password"
                        class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                        id="password"
                        name="password"
                        required=""
                        type="password"
                        value=""
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-yjsfm1"
                        >
                          <span>
                            Password
                             
                            *
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                  <div
                    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="false"
                      for="passwordRepeat"
                      id="passwordRepeat-label"
                    >
                      Repear password
                      <span
                        aria-hidden="true"
                        class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                      >
                         
                        *
                      </span>
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-invalid="false"
                        autocomplete="current-password"
                        class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                        id="passwordRepeat"
                        name="passwordRepeat"
                        required=""
                        type="password"
                        value=""
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-yjsfm1"
                        >
                          <span>
                            Repear password
                             
                            *
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                  <div
                    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-focused Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="true"
                      for="secretCode"
                      id="secretCode-label"
                    >
                      Secret Code
                      <span
                        aria-hidden="true"
                        class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                      >
                         
                        *
                      </span>
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-invalid="false"
                        autocomplete="email"
                        class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                        id="secretCode"
                        name="secretCode"
                        required=""
                        type="text"
                        value=""
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-14lo706"
                        >
                          <span>
                            Secret Code
                             
                            *
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-26yjs-MuiTypography-root"
                  >
                     
                  </p>
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1vhaqj4-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="submit"
                  >
                    <div
                      class="MuiBox-root css-1e16d4q"
                    >
                      Reset Password Up
                    </div>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</main>
`;

exports[`Render should render SignIn screen 1`] = `
<main
  class="MuiContainer-root MuiContainer-maxWidthXs css-nzinn5-MuiContainer-root"
  data-testid="signIn"
>
  <div
    class="MuiBox-root css-binzgt"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1999axt-MuiAvatar-root"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="LockOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
        />
      </svg>
    </div>
    <h1
      class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
    >
      Sign in
    </h1>
    <form
      class="MuiBox-root css-164r41r"
      novalidate=""
    >
      <div
        class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-focused Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for="login"
          id="login-label"
        >
          Email Address
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            autocomplete="email"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="login"
            name="login"
            required=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-14lo706"
            >
              <span>
                Email Address
                 
                *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="password"
          id="password-label"
        >
          Password
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            autocomplete="current-password"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="password"
            name="password"
            required=""
            type="password"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Password
                 
                *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiBox-root css-1b08fi0"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-26yjs-MuiTypography-root"
        >
           
        </p>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1vhaqj4-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="submit"
      >
        <div
          class="MuiBox-root css-1e16d4q"
        >
          Sign In
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
        >
          <a
            class="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineAlways css-101ca9i-MuiTypography-root-MuiLink-root"
            href="auth/passwordReset"
          >
            Forgot password?
          </a>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <a
            class="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineAlways css-101ca9i-MuiTypography-root-MuiLink-root"
          >
            Don't have an account? Sign Up
          </a>
        </div>
      </div>
    </form>
  </div>
</main>
`;

exports[`Render should render SignUp screen 1`] = `
<main
  class="MuiContainer-root MuiContainer-maxWidthXs css-nzinn5-MuiContainer-root"
  data-testid="signUp"
>
  <div
    class="MuiBox-root css-binzgt"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1999axt-MuiAvatar-root"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="LockOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
        />
      </svg>
    </div>
    <h1
      class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
    >
      Sign up
    </h1>
    <form
      class="MuiBox-root css-178yklu"
      novalidate=""
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="login"
              id="login-label"
            >
              Email Address
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                autocomplete="email"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="login"
                name="login"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Email Address
                     
                    *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="password"
              id="password-label"
            >
              Password
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                autocomplete="new-password"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="password"
                name="password"
                required=""
                type="password"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Password
                     
                    *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="passwordRepeat"
              id="passwordRepeat-label"
            >
              Repeat password
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                autocomplete="new-password"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="passwordRepeat"
                name="passwordRepeat"
                required=""
                type="password"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Repeat password
                     
                    *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        />
      </div>
      <div
        class="MuiBox-root css-1b08fi0"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-26yjs-MuiTypography-root"
        >
           
        </p>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1vhaqj4-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="submit"
      >
        <div
          class="MuiBox-root css-1e16d4q"
        >
          Sign Up
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiGrid-root MuiGrid-container css-16g5rpk-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <a
            class="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineAlways css-101ca9i-MuiTypography-root-MuiLink-root"
          >
            Already have an account? Sign in
          </a>
        </div>
      </div>
    </form>
  </div>
</main>
`;
